Foundation/EnergyFormatter.swift

```diff
@@ -177,25 +177,14 @@ open class EnergyFormatter: Formatter {
     /// - Experiment: This is a draft API currently under consideration for official import into Foundation as a suitable alternative
     /// - Note: Since this API is under consideration it may be either removed or revised in the near future
     open override func objectValue(_ string: String) throws -> Any? { return nil }
-}
-
-/// TODO: Replace calls to the below function to use Locale.regionCode
-/// Temporary workaround due to unpopulated Locale attributes
-/// See https://bugs.swift.org/browse/SR-3202
-extension Locale {
-    public var usesCalories: Bool {
-
-        switch self.identifier {
-        case "en_US": return true
-        case "en_US_POSIX": return true
-        case "haw_US": return true
-        case "es_US": return true
-        case "chr_US": return true
-        case "en_GB": return true
-        case "kw_GB": return true
-        case "cy_GB": return true
-        case "gv_GB": return true
-        default: return false
-        }
+    
+    /// Regions that use calories
+    private static let caloriesRegions: Set<String> = ["GB", "US"]
+    
+    /// Whether the region uses calories
+    private var usesCalories: Bool {
+        guard let region = numberFormatter.locale.regionCode else { return false }
+        return EnergyFormatter.caloriesRegions.contains(region)
```

Comment:
This isn't strictly equivalent — regions that aren't `en`, `haw`, `kw`, `cy` or `gv` in language will get calories where they don't elsewhere.

Reply:
I see. So I would go back to Locale.identifier there?